peep.c

```diff
@@ -77,7 +77,7 @@ S_scalar_slice_warning(pTHX_ const OP *o)
         return;
 
     assert(OpSIBLING(kid));
-    name = op_varname(OpSIBLING(kid));
+    name = op_varname(OpSIBLING_nocheck(kid));
```

Comment:
Instead of removing the duplicate check but keeping the duplicate fetch why not use an intermediate variable:
```
    OP *varop = OpSIBLING(kid);
    assert(varop);
    name = op_varname(varop);
```
This avoids the reader having to check both `OpSIBLING(kid)` are the same (and now that `OpSIBLING(kid)` and `OpSIBLING_nocheck(kid)` are the same).